### DISCUSSION TITLE
Dynamic Component

### DISCUSSION THREAD

**Discussion by akil-rails** [vuejs/vitepress#1041] [Q&A] [answered]
The [guide](https://vitepress.vuejs.org/guide/using-vue.html) explains:

> To use code that assumes a browser environment on import, you need to dynamically import them in proper lifecycle hooks: how to use components th

```vue
<script>
export default {
  mounted() {
    import('./lib-that-access-window-on-import').then((module) => {
      // use code
    })
  }
}
</script>
```

When the library is directly referenced (instead of a dependency), the load fails. Is this the expected behaviour? If so, the docs could mention that.

`mermaid:1 Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/html". Strict MIME type checking is enforced for module scripts per HTML spec.`

```vue
<script>
export default {
  mounted() {
    import('https://cdn.skypack.dev/mermaid').then((module) => {
      // use code
    })
  }
}
</script>
```

**Comment by brc-dd** [comment]
I am not getting any error with your code. Can you instead create an issue and give us a reproducible example?

**Reply by akil-rails** [reply]
Thanks , I've created an [issue](https://github.com/vuejs/vitepress/issues/1042) with steps to reproduce, with the example.

**Comment by akil-rails** [comment, marked as answer]
Solved as part of https://github.com/vuejs/vitepress/issues/1042
